### PR TITLE
Feat/add sheet endpoints

### PR DESCRIPTION
**Pull request by LuisOw**
### Testing Instructions

1. Open the PR and review the changes.

### CODE REVIEW

File: pkg/api/handler/sheet_handlers.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	storage *sqlx.DB
+}
+
+func NewSheetHandler(storage *sqlx.DB) *SheetHandler {
+	return &SheetHandler{
+		storage,
+	}
+}
+
+func (sh *SheetHandler) PostSheetHandler(c echo.Context) error {
```
Comment (xsadia) [inline pkg/api/handler/sheet_handlers.go:26]:
```suggestion
func (sh *SheetHandler) CreateSheetHandler(c echo.Context) error {
```

File: pkg/api/handler/sheet_handlers.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+	if err != nil {
+		return err
+	}
+
+	su, erru := sr.Update(sh.storage, os, id)
+
+	if erru != nil {
+		return erru
+	}
```
Comment (xsadia) [inline pkg/api/handler/sheet_handlers.go:110]:
```suggestion
	su, erru := sr.Update(sh.storage, os, id)

	if err != nil {
		return err
	}
```

File: pkg/api/handler/sheet_handlers.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	id, err := uuid.Parse(c.Param("id"))
+
+	if err != nil {
+		return err
+	}
+
+	sr := repository.NewSheetRepository(s)
+	sr.Delete(sh.storage, id)
+
+	return c.NoContent(http.StatusOK)
```
Comment (xsadia) [inline pkg/api/handler/sheet_handlers.go:126]: return http 204 no content

File: pkg/api/handler/test/sheet_handlers_test.go [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+	assert.NoError(suite.T(), errg)
+	assert.Equal(suite.T(), http.StatusOK, rec.Code)
+
+	json.Unmarshal(rec.Body.Bytes(), &jsonRes)
+
+	assert.Equal(suite.T(), jsonRes.Name, sheet.Name)
+	assert.Equal(suite.T(), jsonRes.Description, sheet.Description)
+	assert.Equal(suite.T(), jsonRes.Properties, sheet.Properties)
+	assert.Equal(suite.T(), jsonRes.Background, sheet.Background)
```
Comment (xsadia) [inline pkg/api/handler/test/sheet_handlers_test.go:113]: nitpick create assign `suite.T()` to a variable `t`

Comment (LuisOw) [reply pkg/api/handler/test/sheet_handlers_test.go]: Fixed [5e4b75f](https://github.com/Roll-Play/roll-play-api/pull/5/commits/5e4b75f342dc91f9ec23ef463236b0480337332e)

File: pkg/api/handler/sheet_handlers.go (hunk 2 above)
Comment (LuisOw) [reply pkg/api/handler/sheet_handlers.go]: Fixed [5e4b75f](https://github.com/Roll-Play/roll-play-api/pull/5/commits/5e4b75f342dc91f9ec23ef463236b0480337332e)

File: pkg/api/handler/sheet_handlers.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+func (sh *SheetHandler) GetSheetListHandler(c echo.Context) error {
+	s := new(entities.Sheet)
+
+	if err := c.Bind(s); err != nil {
+		return err
+	}
+
+	p, errp := strconv.Atoi(c.QueryParams().Get("page"))
+	sz, errs := strconv.Atoi(c.QueryParams().Get("size"))
```
[the remark this answers is not in the document]
Comment (LuisOw) [reply pkg/api/handler/sheet_handlers.go]: Fixed [5e4b75f](https://github.com/Roll-Play/roll-play-api/pull/5/commits/5e4b75f342dc91f9ec23ef463236b0480337332e)

File: pkg/api/handler/sheet_handlers.go (hunk 1 above)
Comment (LuisOw) [reply pkg/api/handler/sheet_handlers.go]: Fixed [5e4b75f](https://github.com/Roll-Play/roll-play-api/pull/5/commits/5e4b75f342dc91f9ec23ef463236b0480337332e)

File: pkg/api/handler/sheet_handlers.go (hunk 3 above)
Comment (LuisOw) [reply pkg/api/handler/sheet_handlers.go]: Fixed [5e4b75f](https://github.com/Roll-Play/roll-play-api/pull/5/commits/5e4b75f342dc91f9ec23ef463236b0480337332e) and [074f50d](https://github.com/Roll-Play/roll-play-api/pull/5/commits/074f50db9cccda6d0076c6db211d3527af537c4d)

**Pull request merged**